plugins/inputs/sqlserver/sqlserverqueries.go

```diff
@@ -1209,38 +1222,55 @@ IF SERVERPROPERTY('EngineEdition') NOT IN (2,3,4) BEGIN /*NOT IN Standard,Enterp
 	RETURN
 END
 
-IF SERVERPROPERTY('IsHadrEnabled') = 1 BEGIN
-	SELECT
-		'sqlserver_hadr_dbreplica_states' AS [measurement],
-		REPLACE(@@SERVERNAME,'\',':') AS [sql_instance],
-		database_id,
-		db_name(database_id) as database_name,
-		convert(nvarchar(36), drs.replica_id) as replica_id,
-		ar.replica_server_name,
-		convert(nvarchar(36), drs.group_database_id) as group_database_id,
-		is_primary_replica,
-		synchronization_state,
-		synchronization_state_desc,
-		is_commit_participant,
-		synchronization_health,
-		synchronization_health_desc,
-		database_state,
-		database_state_desc,
-		is_suspended,
-		suspend_reason,
-		suspend_reason_desc,
-		last_sent_time,
-		last_received_time,
-		last_hardened_time,
-		last_redone_time,
-		log_send_queue_size,
-		log_send_rate,
-		redo_queue_size,
-		redo_rate,
-		filestream_send_rate,
-		last_commit_time,
-		secondary_lag_seconds
-	from sys.dm_hadr_database_replica_states AS drs
-	inner join sys.availability_replicas AS ar on drs.replica_id = ar.replica_id
+DECLARE
+	 @SqlStatement AS nvarchar(max)
+	,@MajorMinorVersion AS int = CAST(PARSENAME(CAST(SERVERPROPERTY('ProductVersion') AS nvarchar),4) AS int)*100 + CAST(PARSENAME(CAST(SERVERPROPERTY('ProductVersion') AS nvarchar),3) AS int)
+	,@Columns AS nvarchar(MAX) = ''
+
+IF @MajorMinorVersion >= 1200 BEGIN
+	SET @Columns += N'
+	,is_primary_replica'
 END
+
+IF @MajorMinorVersion >= 1300 BEGIN
```

Comment:
Same as above - 2016 and above versions

Reply:
Added a comment at the start of the query describing the query and the version based specification

Reply:
Please mark all comments as resolved! Thanks!